test/scales/linearScaleTests.ts

```diff
@@ -1,134 +1,267 @@
+        assert.deepEqual(scale.range(), [7, 6], "the range can be unordered");
+      });
+
+      it("maps domain to range in a linear fashion", () => {
+        scale.domain([10, 20]);
+        scale.range([400, 500]);
+        assert.strictEqual(scale.scale(10), 400, "first value in domain maps to first value in range");
+        assert.strictEqual(scale.scale(20), 500, "last value in domain maps to last value in range");
+        assert.strictEqual(scale.scale(15), 450, "middle value in domain maps to middle value in range");
+
+        assert.strictEqual(scale.invert(400), 10, "first value in range maps to first value in domain");
+        assert.strictEqual(scale.invert(500), 20, "last value in range maps to last value in domain");
+        assert.strictEqual(scale.invert(450), 15, "middle value in range maps to middle value in domain");
+
+        scale.domain([20, 10]);
+        assert.strictEqual(scale.scale(10), 500, "first value in flipped domain maps to first value in range");
+        assert.strictEqual(scale.scale(20), 400, "last value in flipped domain maps to last value in range");
+        assert.strictEqual(scale.scale(15), 450, "middle value in flipped domain maps to middle value in range");
+
+        assert.strictEqual(scale.invert(400), 20, "first value in range maps to first value in flipped domain");
+        assert.strictEqual(scale.invert(500), 10, "last value in range maps to last value in flipped domain");
+        assert.strictEqual(scale.invert(450), 15, "middle value in range maps to middle value in flipped domain");
+      });
 
-    it("domainMin() and domainMax() together", () => {
-      let scale = new Plottable.Scales.Linear();
-      scale.padProportion(0);
-      let requestedDomain = [-5, 5];
-      scale.addIncludedValuesProvider((scale: Plottable.Scales.Linear) => requestedDomain);
-
-      let desiredMin = -10;
-      let desiredMax = 10;
-      scale.domainMin(desiredMin);
-      scale.domainMax(desiredMax);
-      assert.deepEqual(scale.domain(), [desiredMin, desiredMax], "setting domainMin() and domainMax() sets the domain");
-
-      scale.autoDomain();
-      let bigMin = 10;
-      let smallMax = -10;
-      scale.domainMin(bigMin);
-      scale.domainMax(smallMax);
-      assert.deepEqual(scale.domain(), [bigMin, smallMax], "setting both is allowed even if it reverse the domain");
+      it("filters out invalid numbers when using extentOfValues()", () => {
+        let arrayWithBadValues: any[] = [null, NaN, undefined, Infinity, -Infinity, "a string", 1, 1.2];
+        let expectedExtent = [1, 1.2];
+        let extent = scale.extentOfValues(arrayWithBadValues);
+        assert.deepEqual(extent, expectedExtent, "invalid values were filtered out");
+      });
+
+      it("does not accept NaN or Infinity as domain", () => {
+        let warningCalled = false;
+        let oldWarn = Plottable.Utils.Window.warn;
+        Plottable.Utils.Window.warn = (msg: string) => {
```

Comment:
If this is common enough in tests, maybe we should write `TestMethod.assert.warns()`?